hexrd/ui/main_window.py

```diff
@@ -718,6 +718,18 @@ def on_action_edit_euler_angle_convention(self):
     def on_action_edit_apply_hand_drawn_mask_triggered(self):
         # Make the dialog
         self._apply_drawn_mask_line_picker = (HandDrawnMaskDialog(self.ui))
+        self.ui.image_tab_widget.tabBarClicked.connect(
+            self._apply_drawn_mask_line_picker.tabbed_view_changed)
+        HexrdConfig().tab_images_changed.connect(
+            lambda: self._apply_drawn_mask_line_picker.tabbed_mode_changed(
+                self.ui.image_tab_widget.active_canvases
+            )
+        )
```

Comment:
I think I have an idea that will simplify the design.

What if we let the `HandDrawnMaskDialog` only hold on to a reference to a single canvas: the one that is currently active? It can then have a `change_canvas()` function that switches the canvas to a different one (it will call `mpl_disconnect()` for the old canvas and `mpl_connect()` for the new canvas).

Then, we just need to keep track somewhere in the code when the active canvas changes. Maybe the best place to do that is in the image tab widget? It can emit a signal anytime the active canvas changes. Then this main window function can call `change_canvas()` on the `HandDrawnMaskDialog`. We could also probably use this signal from the image tab widget that the active canvas changed in other places in the code too.

Here is my current understanding about the active canvas: 

1. If the user does not change to the tabbed view in raw mode, but switches between view modes (cartesian, polar, etc.), then the active canvas does not change. 
2. But if the user changes tabs in the tabbed raw view, then the active canvas changes. And if they then switch tabs or from a non-first tab to a different view mode, the active canvas will change again.
3. It looks like the non-first tab canvases get deleted and remade for every rerender (maybe we should redesign it so that isn't the case). But right now, if the user is on a non-first tab canvas, and a rerender happens, then the active canvas changes (because the old one was deleted and a new one was made).

We would need to take into account that information when emitting the signal when the active canvas changes. But I think it will be easier for the image tab widget to keep track of which canvas is active, and emit a signal when the active canvas changes.

Reply:
So I did actually try this approach (only tracking the active canvas since that is all we need) but I was encountering the same issues as with the list of active canvases. Your breakdown pointed me at the real culprit though:

> It looks like the non-first tab canvases get deleted and remade for every rerender (maybe we should redesign it so that isn't the case). But right now, if the user is on a non-first tab canvas, and a rerender happens, then the active canvas changes (because the old one was deleted and a new one was made).

I did not catch this initially! Basically what was happening was the connections were made to the new canvases, but then any existing in-progress masks would be completed (the raw masks would be updated) which would trigger a re-render and new canvases and the connections that were created would be useless. So thanks for catching this!

This has been cleaned up now and we now have an `active_canvas_changed` function in the main window that can be used for anything else that may need to be updated when the active canvas changes.